Embedded and interface build error, `Interface_xxx_method_wrapper undeclared`

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.2.4 d8cce0a** 
**OS: ArchLinux** 

**What did you do?**
```v
fn main(){
	mut ll := &LinearLayout{}
	mut lbl := &Label{
		x: 10
		y: 20
	}
	ll.add(mut lbl)  //  1) works if comment out
}

//-----------------------

pub struct Rect{
mut:
	x      int
	y      int
}

pub fn (r Rect) get_pos() (int, int) {
	return r.x, r.y
}

//------------------------

pub struct LayoutBase{
	Rect
}

//------------------------

pub struct Base{
	LayoutBase
}

pub fn (mut b Base) init(){}    // 2) works if comment out, correct error message.

//------------------------

[heap]
pub struct Label{
	Base
}

//------------------------

pub interface Layoutable{
	get_pos() (int, int)
mut:               // 3-a) works if comment out
	init()     // 3-b) 
}

[heap]
pub struct LinearLayout{
	Base
mut:
	layoutables []Layoutable 
}

pub fn (mut ll LinearLayout) add(mut l Layoutable){}
```

**What did you expect to see?**
No error.
 
**What did you see instead?**
```
> v run maintest.v
==================
/tmp/v_1000/maintest.7390342816253743358.tmp.c:1916: error: 'main__Rect_get_pos_Interface_main__Layoutable_method_wrapper' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -3604,10 +3604,15 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 	}
 	// struct embedding
 	if sym.info in [ast.Struct, ast.Aggregate] {
-		for embed in node.from_embed_types {
+		for i, embed in node.from_embed_types {
 			embed_sym := g.table.sym(embed)
 			embed_name := embed_sym.embed_name()
-			if node.expr_type.is_ptr() {
+			is_left_ptr := if i == 0 {
+				node.expr_type.is_ptr()
+			} else {
+				node.from_embed_types[i - 1].is_ptr()
+			}
+			if is_left_ptr {
 				g.write('->')
 			} else {
 				g.write('.')
@@ -6831,6 +6836,7 @@ static inline __shared__$interface_name ${shared_fn_name}(__shared__$cctype* x) 
 				styp := g.cc_type(method.params[0].typ, true)
 				mut method_call := '${styp}_$name'
 				if !method.params[0].typ.is_ptr() {
+					method_call = '${cctype}_$name'
 					// inline void Cat_speak_Interface_Animal_method_wrapper(Cat c) { return Cat_speak(*c); }
 					iwpostfix := '_Interface_${interface_name}_method_wrapper'
 					methods_wrapper.write_string('static inline ${g.typ(method.return_type)} ${cctype}_$name${iwpostfix}(')

--- a/vlib/v/tests/interface_with_multi_nested_embed_1_test.v
+++ b/vlib/v/tests/interface_with_multi_nested_embed_1_test.v
@@ -1,4 +1,4 @@
-fn test_interface_struct_with_multi_nested_embed() {
+fn test_interface_with_multi_nested_embed() {
 	mut win := &Window{}
 	mut ll := &LinearLayout{}
 	mut lbl := &Label{

--- a/vlib/v/tests/interface_with_multi_nested_embed_2_test.v
+++ b/vlib/v/tests/interface_with_multi_nested_embed_2_test.v
@@ -1,0 +1,57 @@
+fn test_interface_with_multi_nested_embed() {
+	mut ll := &LinearLayout{}
+	mut lbl := &Label{
+		x: 10
+		y: 20
+	}
+	ll.add(mut lbl)
+
+	println(ll)
+
+	assert ll.x == 10
+	assert ll.y == 20
+}
+
+pub struct Rect {
+mut:
+	x int
+	y int
+}
+
+pub fn (r Rect) get_pos() (int, int) {
+	return r.x, r.y
+}
+
+pub struct LayoutBase {
+	Rect
+}
+
+pub struct Base {
+	LayoutBase
+}
+
+pub fn (mut b Base) init() {}
+
+[heap]
+pub struct Label {
+	Base
+}
+
+pub interface Layoutable {
+	get_pos() (int, int)
+mut:
+	init()
+}
+
+[heap]
+pub struct LinearLayout {
+	Base
+mut:
+	layoutables []Layoutable
+}
+
+pub fn (mut ll LinearLayout) add(mut l Layoutable) {
+	x, y := l.get_pos()
+	ll.x += x
+	ll.y += y
+}

